### PR TITLE
Add flake8 linting.

### PR DESCRIPTION
**Pull request by tomprince**
Here are the issues that were reported by flake8, along with some comments on how I addressed them where appropriate.
- E402 module level import not at top of file
  This was cause by some code calling `registerAdapter` on `ZKAPAuthorizerStorageServer`. It looks like that was needed because the interfaces for it were defined by `implementer_only` which explicitly ignored interfaces defined by parent classes. Looking at the history, I assume the use of `implementer_only` was to avoid implementing `RIStorageServer` via `proxyForInterface`. Given that the later is no longer used, `implementer_only` is no longer necessary.

  Reading some documentation, I think an alternative to needing to having the `registerAdapter` would have been to use
  ```patch
  -@implementer_only(RIPrivacyPassAuthorizedStorageServer, IReferenceable, IRemotelyCallable)
  +@implementer_only(RIPrivacyPassAuthorizedStorageServer, implementedBy(Referenceable))
  ```

- E722 do not use bare 'except'
  These were replaced by `except Exception:`
- E302 expected 2 blank lines, found 1
- E501 line too long 
  These are the changes I'm least sure of. I did a variety of things for this
  - Most places, there were reasonable ways to wrap the code, which I did.
  - There was one place where black didn't like wrapping a line with a `lambda` with a bunch of parameters. Here I changed the order of agruments, and converted the lambda to use `partial` instead. 
  - There were three places where there wasn't a good place to wrap, which I annotated with `# noqa: E501`
- E731 do not assign a lambda expression, use a def
  - I disabled this error in `*/tests/*.py`.

### CODE REVIEW

File: src/_zkapauthorizer/schema.py [hunk 1]
```diff
@@ -77,15 +74,15 @@ def run_schema_upgrades(upgrades, cursor):
         """
         CREATE TABLE [vouchers] (
             [number] text,
-            [created] text,                     -- An ISO8601 date+time string.
```
Comment (exarkun) [inline src/_zkapauthorizer/schema.py:80]: Hm.  These text changes are all semantically null so they should be fine.  However, SQLite3 stores the schema as text in the database and as a result I'm vaguely afraid of changing the exact text of the old versions of the schema.

So I read https://sqlite.org/fileformat2.html#ffschema and I'm less afraid now.  The text is stored but of course SQLite3 has to parse it to actually do anything with it so non-semantic changes should make no difference to operation.  Further, SQLite3 already performs certain normalization to the text before storing it.  At first I thought it collapsed all whitespace in the statement down to a single space but on experimentation I see that it really only does this for whitespace between "create table" and the table name - not any subsequent whitespace.  Still, it seems the difference in the text should be harmless.

The moved comments are slightly different but it seems like they should likewise be harmless.

So, okay, no adjustment required here.

**Review by exarkun — CHANGES_REQUESTED** [review]
Thanks.  I think I looked at all the changes distinct to this branch.  Some comments inline.  In particular, I'm not convinced the `except` to `except Exception` change is safe/correct.  Handling `BaseException` in all those places would at least be the _same_ as it was before (whether it was 100% correct before or not could be a separate question of course).

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[flake8]
+max-line-length = 88
+extend-ignore = E203
```
[the remark this answers is not in the document]
Comment (tomprince) [reply .flake8:3]: No. This is the suggested black-compatible configuration (per https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8). I've added a comment.

File: src/_zkapauthorizer/tests/_exception.py [hunk 3]
```diff
@@ -94,10 +89,10 @@ def __init__(self, exception_matcher):
     def match(self, matchee):
         try:
             result = matchee()
-            return Mismatch('%r returned %r' % (matchee, result))
+            return Mismatch("%r returned %r" % (matchee, result))
         # Catch all exceptions: Raises() should be able to match a
         # KeyboardInterrupt or SystemExit.
-        except:
+        except Exception:
```
[the remark this answers is not in the document]
Comment (tomprince) [reply src/_zkapauthorizer/tests/_exception.py:95]: It looks like this matcher is available in testtools now, so I've removed this module and adjusted the imports.

[the remark this answers is not in the document]
Comment (exarkun) [reply src/_zkapauthorizer/tests/_exception.py:95]: I have a vague memory of the one in testtools doing worse error reporting.

**Comment by tomprince** [comment]
I'm going to close this for the moment. Getting black and isort landed was the part that was more significant to me. I'm not sure the changes here are worth the effort, and I don't seem to have the right kind of attention to detail make the changes here at the moment. I will split out the changes to remove the need for `registerAdapter`, since I believe that part doesn't have any issues.

**Pull request closed without merging**